File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Models/Story.swift
import Foundation

public struct Story: Identifiable, Codable {
    public let id: String
    public let userId: String
    public let userName: String
    public let userProfileImage: String?
    public let mediaUrl: String
    public let mediaType: StoryMediaType
    public let duration: Double?
    public let thumbnailUrl: String?
    public let views: Int
    public let viewedBy: [String]?
    public let isActive: Bool
    public let createdAt: String
    public let expiresAt: String

    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case userId, userName, userProfileImage, mediaUrl, mediaType, duration, thumbnailUrl, views,
            viewedBy, isActive, createdAt, expiresAt
    }
}

public enum StoryMediaType: String, Codable {
    case image
    case video
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/data/model/Story.kt
package com.localforvocalstartup.app.data.model

import java.util.Date

data class Story(
    val _id: String,
    val userId: String,
    val userName: String,
    val userProfileImage: String?,
    val mediaUrl: String,
    val mediaType: StoryMediaType,
    val duration: Double?,
    val thumbnailUrl: String?,
    val views: Int,
    val viewedBy: List<String>?,
    val isActive: Boolean,
    val createdAt: String,
    val expiresAt: String
)

enum class StoryMediaType {
    image,
    video
}
